### PR TITLE
Add transaction expiration

### PR DESCRIPTION
**Pull request by belliottsmith**

### CODE REVIEW

File: accord-core/src/main/java/accord/impl/InMemoryCommandStore.java [hunk 1]
```diff
@@ -163,7 +157,17 @@ public long latestEpoch()
         }
 
         @Override
-        public Timestamp maxConflict(Keys keys)
+        public Timestamp preaccept(TxnId txnId, Keys keys)
+        {
+            Timestamp max = maxConflict(keys);
+            long epoch = latestEpoch();
+            if (txnId.compareTo(max) > 0 && txnId.epoch >= epoch && !agent.isExpired(txnId, time.now()))
```
Comment (aweisberg) [inline accord-core/src/main/java/accord/impl/InMemoryCommandStore.java:164]: Why return a new timestamp if is expired? I thought the expiration check is done on the coordinator now?

**Review by aweisberg — CHANGES_REQUESTED** [review]
Only the one question. Wondering if that `isExpired` check in `InMemoryCommandStore` is left over from when expiration sent back the max ballot.

File: accord-core/src/main/java/accord/impl/InMemoryCommandStore.java (hunk 1 above)
Comment (belliottsmith) [reply accord-core/src/main/java/accord/impl/InMemoryCommandStore.java:164]: This permits a majority of remote replicas to also collectively reject an expired transaction received from a long running coordinator  - and also means that recovery doesn't accidentally register a preaccept that can be counted towards the original coordination. Though perhaps for that we should just explicitly refuse to use the original ballot for recovery.

This is all sort of arbitrary, so happy with any of these approaches.

**Review by aweisberg — APPROVED** [review]
+1 TY

**Pull request merged**